Sources/MapboxMaps/Gestures/GestureHandlers/SingleTapGestureHandler.swift

```diff
@@ -0,0 +1,27 @@
+import UIKit
+
+/// `SingleTapGestureHandler` manages a gesture recognizer looking for single tap touch events
+internal final class SingleTapGestureHandler: GestureHandler {
+
+    internal init(gestureRecognizer: UITapGestureRecognizer,
+                  mapboxMap: MapboxMapProtocol,
```

Comment:
I'm feeling less good about the existence of this handler again. It's fundamentally different than the other gestures since it doesn't manipulate the map, but still depends on mapboxMap since that's a requirement of the base class.

Reply:
Yeah this part bothers me as well. I'd love to move away from GestureHandler being a super class to be honest. It feels like it'd be better suited to be a protocol.

Reply:
okay. let's leave it for now. we can refactor it later